R/bundleFiles.R

```diff
@@ -174,19 +174,28 @@ ignoreBundleFiles <- function(dir, contents, depth = 0) {
   # exclude some well-known files/directories at root level
   if (depth == 0) {
     contents <- contents[!grepl(glob2rx("*.Rproj"), contents)]
-    contents <- setdiff(
-      contents,
-      c("manifest.json", "rsconnect", "packrat", "app_cache", ".Rproj.user")
-    )
+    contents <- setdiff(contents, c("manifest.json", "app_cache", ".Rproj.user"))
   }
 
-  # exclude renv files, knitr cache dirs, and another well-known files
-  contents <- setdiff(contents, c("renv", "renv.lock"))
-  contents <- contents[!isKnitrCacheDir(contents)]
-  contents <- setdiff(
-    contents,
-    c(".DS_Store", ".git", ".gitignore", ".quarto", ".Rhistory", ".svn")
+  ignore <- c(
```

Comment:
Is it intentional that a file like `.Rproj.user` is ignored only at the root while `.Rhistory` is ignored throughout the hierarchy? I don't recall why we differentiate. I'd like to understand the reasoning.

Reply:
I don't know either; I'd be in favour of just ignoring everything at every level.

Reply:
The initial implementation only ignored at the root: https://github.com/rstudio/rsconnect/pull/112
renv files were ignored at all levels: https://github.com/rstudio/rsconnect/pull/367

I think I'm also inclined to ignore throughout the hierarchy, since we cannot assume the appDir doesn't have complicated directory structures beneath. NEWS the change?